03-auto_tick.xsh

```diff
@@ -2,303 +2,90 @@
+        pull_request=True, rerender=True, fork=True, gh=None,
+        **kwargs):
+    # get the repo
+    feedstock_dir = get_repo(feedstock=feedstock, protocol=protocol,
+             pull_request=pull_request, fork=fork, gh=gh)
 
-    feedstock_dir = os.path.join($REVER_DIR, $PROJECT + '-feedstock')
+    # migrate the `meta.yaml`
     recipe_dir = os.path.join(feedstock_dir, 'recipe')
-    if not os.path.isdir(feedstock_dir):
-        p = ![git clone @(origin) @(feedstock_dir)]
-        if p.rtn != 0:
-            msg = 'Could not clone ' + origin
-            msg += '. Do you have a personal fork of the feedstock?'
-            return
-    with indir(feedstock_dir):
-        git fetch @(origin)
-        # make sure feedstock is up-to-date with origin
-        git checkout master
-        git pull @(origin) master
-        # make sure feedstock is up-to-date with upstream
-        git pull @(upstream) master
-        # make and modify version branch
-        with ${...}.swap(RAISE_SUBPROC_ERROR=False):
-            git checkout $VERSION or git checkout -b $VERSION master
-    # Render with new version but nothing else
-    with indir(recipe_dir):
-        for f, p, n in patterns:
-            p = eval_version(p)
-            n = eval_version(n)
-            replace_in_file(p, n, f)
-        with open('meta.yaml', 'r') as f:
-            text = f.read()
-        # If we can't parse the meta_yaml then jump out
-        meta_yaml = parsed_meta_yaml(text)
-        # If the parser returns None, then we didn't read the meta.yaml
-        # TODO: How we didn't fail at 01 on this recipe is mysterious
-        if meta_yaml is None:
-            with open('upstream_bad', 'a') as f:
-                f.write('{}: failed to read meta.yaml\n'.format($PROJECT))
-            rm -rf @(feedstock_dir)
-            return False
-        source_url = meta_yaml.get('source', {}).get('url')
-        if not source_url:
-            with open('upstream_bad', 'a') as f:
-                f.write('{}: missing url\n'.format($PROJECT))
-                rm -rf @(feedstock_dir)
-                return False
-        if isinstance(source_url, list):
-            for url in source_url:
-                if 'Archive' not in url:
-                    source_url = url
-                    break
-        if 'cran.r-project.org/src/contrib' in source_url:
-            $VERSION = $VERSION.replace('_', '-')
-
-    # now, update the feedstock to the new version
-    source_url = eval_version(source_url)
-    try:
-        hash = hash_url(source_url, hash_type)
-    except urllib.error.HTTPError:
-        with open('upstream_bad', 'a') as f:
-            t = '{}: hash failed at {}\n'.format(
-                meta_yaml.get('package', {}).get('name', 'UNKOWN'), source_url)
-            print(t)
-            f.write(t)
-        rm -rf @(feedstock_dir)
+    if not migrator.migrate(recipe_dir, attrs, **kwargs):
+        rm - rf @ (feedstock_dir)
```

Comment:
This seems to have weird spacing.  Should it be `rm -rf @(feedstock_dir)`

Reply:
pycharm strikes again! (One of these days I'm going to write a pycharm xonsh extension).

Reply:
haha, it may be easier to just write your own editor...just saying :wink: